File: frontend/src/model/TrackerAnalyticsResponseDto.ts
import {
    NumericChartAnalyticResultDto,
    ScatterChartAnalyticResultDto,
    SingleValueAnalyticResultDto,
} from "./AnalyticResultDto";

export interface TrackerAnalyticsResponseDto {
    singleValueAnalytics: SingleValueAnalyticResultDto[];
    numericChartAnalytics: NumericChartAnalyticResultDto[];
    scatterChartAnalytics: ScatterChartAnalyticResultDto[];
}

File: backend/src/Operum.Model/DTOs/Trackers/TrackerAnalyticsResponseDto.cs
using Operum.Model.DTOs.Analytics;

namespace Operum.Model.DTOs.Trackers
{
    public class TrackerAnalyticsResponseDto
    {
        public List<SingleValueAnalyticResult> SingleValueAnalytics { get; set; } = [];
        public List<NumericChartAnalyticResult> NumericChartAnalytics { get; set; } = [];
        public List<ScatterChartAnalyticResult> ScatterChartAnalytics { get; set; } = [];
    }
}
